init.js

```diff
@@ -154,6 +132,34 @@ require.cache[SHMPath].exports = class WebSocketManager extends SHM {
 				this.reconnect();
 			});
 
+			const hotReload = this.client.options.hotReload;
+			if(hotReload && shard.sessionID) {
+				shard.once(Constants.ShardEvents.RESUMED, () => {
+					this.debug("Shard session resumed. Restoring cache", shard);
+					shard.loadCacheTimeout = null;
+					this.client.clearTimeout(this.loadCacheTimeout);
+					const cache = hotReload.cacheData;
+					if(cache?.guilds) {
+						for(const [id, guild] of Object.entries(cache.guilds)) {
+							if(ShardClientUtil.shardIDForGuildID(id, this.totalShards) === shard.id) {
+								this.client.guilds.add(guild);
+							}
+						}
+					} else {
+						const { guilds } = this.client._loadCache("guilds", id => ShardClientUtil.shardIDForGuildID(id, this.totalShards) === shard.id);
+						for(const guild of Object.values(guilds)) {
+							this.client.guilds.add(guild);
+						}
+					}
+				})
+
+				shard.loadCacheTimeout = this.client.setTimeout(() => {
+					this.debug("Shard cache was never loaded as the session didn't resume in 15s", shard);
+					shard.loadCacheTimeout = null;
+					shard.removeEventListener(Constants.ShardEvents.RESUMED) // Remove the event in a better way?
```

Comment:
What is the preferred way to remove the event?

Reply:
https://github.com/timotejroiko/discord.js-light/pull/41/commits/818d0e9d8797e75a065a785ea2fefe443c86199a Fixes errors and it seems to work. Double check to make sure it is the correct way to do things

Reply:
if we join the session loading and cache loading then we dont need to remove the event, we add it only if there is a session load